azure-pipelines-code-mirror.yml

```
@@ -43,6 +43,36 @@ jobs:
         displayName: Hard reset local branch to GitHub branch
         workingDirectory: $(WorkingDirectoryName)
       - script: |
+          set /a retryattempt=0
+          :PUSH_RETRY
           git push azdo-mirror $(BranchToMirror) $(ExtraPushArgs)
+          @echo git push azdo-mirror BranchToMirror ExtraPushArgs
+
+          if %errorlevel% NEQ 0 (
+              echo Push failed
+              REM Retry up to three times
+              if %retryattempt% GEQ 3 (
+                  goto :PUSH_FAILURE
+              )
+
+              set /a retryattempt+=1
+              @echo Retry attempt %retryattempt% of 3 in 5 seconds...
+              ping 127.0.0.1 -n 6 > nul
```

as Alex noted Timeout 6 is fine too.

---

> but in both cases 6 pings = 6 seconds, not 5, no?

It's a fence-post. The first count for ping happens immediately, only then does it pause. So it needs `n+1`.

---

Some research indicated that `timeout` interacts with the terminal. It can continue immediately on user-press (which we don't expect) and it can fail hard if headless (which may happen?). Using ping seems the safest.

---

ah that's what I had thought previously :)  carry on.